src/methods/lombscargle.jl

```diff
@@ -120,8 +120,7 @@ function (sg::SurrogateGenerator{<:IrregularLombScargle})()
 end
 
 function swap_elements!(candidate, s, rng, range)
-    k = sample(rng, range)
-    l = sample(rng, range)
+    k, l = sample(rng, range, 2, replace = false)
```

Comment:
It is actually better to use teh previous version for performance. `sample` here allocates a 2-element vector :)

Reply:
```
julia> x = rand(100);

julia> sample1(x) = sample(x, 2; replace = false);

julia> sample2(x) = (a = sample(x); b = sample(x); return (a, b))
sample2 (generic function with 1 method)

julia> sample1(x)
2-element Vector{Float64}:
 0.12229836237241765
 0.0943741108705125

julia> sample2(x)
(0.9856096085646662, 0.4488287731030546)

julia> using BenchmarkTools

julia> @btime sample1($x);
  41.532 ns (1 allocation: 80 bytes)

julia> @btime sample2($x);
  17.836 ns (0 allocations: 0 bytes)
```

Reply:
Good catch. I wanted to sample both at the same time, so I could make sure I sampled without replacement (so that we don't swap the same elements), but that isn't necessary, because it will occur very rarely, and won't affect the result of the algorithm.